src/loaders/gtdb/gtdb_taxa_freq_loader.py

```diff
@@ -71,9 +71,38 @@ def _create_docs(nodes, load_version, kbase_collection):
                 "name": name,
                 "count": nodes[rank][name]
             }
-            docs.append(doc)
+            freq_docs.append(doc)
+
+        identical_ranks.add(rank)
+
+    return freq_docs, identical_ranks
+
+
+def _create_rank_docs(kbase_collection, load_version, identical_ranks):
+    rank_candidates = list(_TAXA_TYPES.values())
+    rank_selected = [False] * len(rank_candidates)
+
+    for identical_rank in identical_ranks:
+        idx = rank_candidates.index(identical_rank)
+        rank_selected[idx] = True
+
+    rank_doc = [{
+        "_key": hashlib.md5(
+            f"{kbase_collection}_{load_version}".encode('utf-8')
+        ).hexdigest(),
+        "collection": kbase_collection,
+        "load_version": load_version,
+        "ranks": [rank_candidates[i] for i in range(len(rank_selected)) if rank_selected[i]]}]
+
+    return rank_doc
+
+
+def _create_docs(nodes, load_version, kbase_collection):
```

Comment:
I would just inline this method given how short it is

Reply:
👍

Reply:
:+1: